android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt

```diff
@@ -0,0 +1,86 @@
+package com.ackee.flipper.grpc
+
+import com.google.gson.GsonBuilder
+import io.grpc.*
+import java.util.*
+
+/**
+ * Interceptor that logs requests and responses from gRPC server
+ */
+internal class FlipperGrpcInterceptor(
+    private val flipperGrpcPlugin: FlipperGrpcPlugin
+) : ClientInterceptor {
+
+    override fun <ReqT : Any?, RespT : Any?> interceptCall(
+        method: MethodDescriptor<ReqT, RespT>,
+        callOptions: CallOptions,
+        next: Channel
+    ): ClientCall<ReqT, RespT> {
+        val id = UUID.randomUUID().toString()
+
+        return object : ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
+            next.newCall(
+                method,
+                callOptions
+            )
+        ) {
+
+            private var headers: Metadata? = null
+
+            override fun start(responseListener: ClientCall.Listener<RespT>, headers: Metadata) {
+                this.headers = headers
+                class LoggingClientCallListener :
```

Comment:
I did not even know you can do this like in Java :D Wouldn't it be better to move it outside of the method body? :D

Reply:
c10d37fc064ea28596684833d493db71120b54b3